pkg/provider/k8s/k8s.go

```diff
@@ -31,23 +31,140 @@ type K8s struct {
 	ctx context.Context
 }
 
+// K8sClient holds the fields used to generate API reqeust from within a cluster.
+type K8sClient struct {
+	// The k8s provider used when we work with the manifest files.
+	k8sProvider *K8s
+	// DeploymentFiles files provided from the cli.
+	DeploymentFiles []string
+	// Vaiables to subtitude in the DeploymentFiles.
+	// These are also used when the command requires some variables that are not provided by the deployment file.
+	DeploymentVars map[string]string
+	// DeploymentFile content after substituting the variables filename is used as the map key.
+	deploymentsContent []provider.ResourceFile
+
+	ctx context.Context
+}
+
 // New returns a k8s client that can apply and delete resources.
-func New(ctx context.Context, config clientcmdapi.Config) (*K8s, error) {
-	restConfig, err := clientcmd.NewDefaultClientConfig(config, &clientcmd.ConfigOverrides{}).ClientConfig()
+func New(ctx context.Context, clientset *kubernetes.Clientset) *K8s {
```

Comment:
Can't we  keep it as is and if `config ==nil' than initialise using the `InClusterConfig` ?

Reply:
Yeah using default arguments is a good choice